nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/coldstorage/CheckAvailabilityOfColdStorageContentListener.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *  
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.core.coldstorage;
+
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ecm.core.api.CloseableCoreSession;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.blob.ColdStorageHelper;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventListener;
+import org.nuxeo.ecm.core.repository.RepositoryService;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * An asynchronous listener that checks if the contents being retrieved form cold storage are available.
```

Comment:
I believe you want a **synchronous** listener right? Since it's triggered by a scheduler that is already asynchronous.
@efge can you please confirm?

In fact, since it's implementing `EventListener`, it is synchronous, no matter what attributes you put in the listener's contribution.

Reply:
i see, i added this comment because i defined it in the xml contribution as below:

`<listener name="checkAvailabilityOfBlobsBeingRetrieved" async="true"`

`async="true"`

Reply:
i see i added because i define my listener as:

```
<listener name="checkAvailabilityOfBlobsBeingRetrieved" async="true"...
```

`async="true".`